src/http/Message.h

```diff
@@ -141,24 +146,30 @@ class Message : public RefCountable
     bool parseHeader(Http1::Parser &, Http::ContentLengthInterpreter &); // TODO move this function to the parser
 };
 
-} // namespace Http
-
-inline void
-HTTPMSGUNLOCK(Http::Message *a)
+inline bool
+Message::Destroy(Message *m)
```

Comment:
Is there a C++ trick that would allow us to have one (possibly templated) definition of this static method while failing the compilation if the method is given anything other than a **raw** pointer to a Message class or a Message class child (e.g., a smart pointer to a Message class)?

Reply:
My concern is code duplication -- adding multiple essentially identical HTTPMSGUNLOCK() methods, one for each message type. If we can somehow make this Destroy() method fail for smart pointers while allowing raw pointers, we can avoid that duplication.

Reply:
Done with static_assert.